### PR TITLE
chore(apigatewayv2): review readme

### PR DESCRIPTION
**Pull request by sumupitchayan**
API Review Session for `aws-apigatewayv2-alpha`, `aws-apigatewayv2-authorizers-alpha`, and `aws-apigatewayv2-integrations-alpha`.

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/aws-apigatewayv2-alpha/README.md [hunk 1]
```diff
@@ -46,6 +46,8 @@ This module supports features under [API Gateway v2](https://docs.aws.amazon.com
 that lets users set up Websocket and HTTP APIs.
 REST APIs can be created using the `aws-cdk-lib/aws-apigateway` module.
 
+HTTP and Websocket APIs use the same Cloudformation resources under the hood. However, this module separates them into two separate constructs for a more efficient abstraction since there are a number of Cloudformation properties that specifically apply only to each type of API.
```
Comment (mikewrighton) [inline packages/@aws-cdk/aws-apigatewayv2-alpha/README.md:49]: CloudFormation

File: packages/@aws-cdk/aws-apigatewayv2-integrations-alpha/README.md
Comment (mikewrighton) [inline packages/@aws-cdk/aws-apigatewayv2-integrations-alpha/README.md:1]: Everywhere: lambda -> Lambda

File: packages/@aws-cdk/aws-apigatewayv2-authorizers-alpha/README.md [hunk 2]
```diff
@@ -39,9 +39,9 @@ API](https://docs.aws.amazon.com/apigateway/latest/developerguide/http-api-acces
 
 ## HTTP APIs
 
-Access control for Http Apis is managed by restricting which routes can be invoked via.
+Access control for Http APIs is managed by restricting which routes can be invoked via.
```
Comment (vinayak-kukreja) [inline packages/@aws-cdk/aws-apigatewayv2-authorizers-alpha/README.md:42]: NIT: We can write `HTTP`

File: packages/@aws-cdk/aws-apigatewayv2-integrations-alpha/README.md
Comment (mikewrighton) [inline packages/@aws-cdk/aws-apigatewayv2-integrations-alpha/README.md:1]: `The details on this format is available` -> `The details on this format are available`

Comment (mikewrighton) [inline packages/@aws-cdk/aws-apigatewayv2-integrations-alpha/README.md:1]: `When a client connects/disconnects or sends message specific to a route` -> `When a client connects/disconnects or sends a message specific to a route`

Comment (mikewrighton) [inline packages/@aws-cdk/aws-apigatewayv2-integrations-alpha/README.md:1]: `sendmessage` -> `sendMessage`?

File: packages/@aws-cdk/aws-apigatewayv2-authorizers-alpha/README.md
Comment (mikewrighton) [inline packages/@aws-cdk/aws-apigatewayv2-authorizers-alpha/README.md:1]: `Lambda Authorizers, JWT authorizers and standard AWS IAM roles` -> `Lambda Authorizers, JWT authorizers, and standard AWS IAM roles`

Oxford commas FTW

Comment (mikewrighton) [inline packages/@aws-cdk/aws-apigatewayv2-authorizers-alpha/README.md:1]: Everywhere: Http -> HTTP

Comment (mikewrighton) [inline packages/@aws-cdk/aws-apigatewayv2-authorizers-alpha/README.md:1]: Incomplete sentence? `Access control for Http APIs is managed by restricting which routes can be invoked via.`

Comment (mikewrighton) [inline packages/@aws-cdk/aws-apigatewayv2-authorizers-alpha/README.md:1]: Everywhere: api -> API

Comment (mikewrighton) [inline packages/@aws-cdk/aws-apigatewayv2-authorizers-alpha/README.md:1]: `Authorization can also configured for each Route` -> `Authorization can also be configured for each Route`

Comment (mikewrighton) [inline packages/@aws-cdk/aws-apigatewayv2-authorizers-alpha/README.md:1]: will require the [write:books] scope -> will require the `['write:books']` scope

Comment (mikewrighton) [inline packages/@aws-cdk/aws-apigatewayv2-authorizers-alpha/README.md:1]: `You can set an authorizer to your WebSocket API's` -> `You can set an authorizer to your WebSocket APIs`

Comment (vinayak-kukreja) [inline packages/@aws-cdk/aws-apigatewayv2-authorizers-alpha/README.md:1]: > You can set an authorizer to your WebSocket API's $connect route to control access to your API.

Does this also apply to Lambda Authorizers in WebSocket? Or do we have a property that reflects this?

**Review by scanlonp — COMMENTED** [review]
General comments:

- We do not explain integrations before we use them in ApiGatewayV2. I do not think linking the other module is enough. We should give a quick run down of what the integrations are used for, and how they help before we use them in the first code example.
- Using routes is not explained in detail. I see many examples of adding routes, but few on using them after creation.b
  - In the authorizers readme, it does a grant method on `routes[0]` (line 225)
  - Is this how we should reference routes? What if I have 10 routes, do I need to keep track of the array? Could I give them a name instead to reference by?
- IAM authorizers are not consistent between HTTP and Websocket.
  - In the websocket section, we create an IAM user and manually write a policy. This cannot be the preferred way of adding IAM policys (288)
- I would like to see more in-line comments in the code examples, especially the early examples. There is too much unexplained code. I think a short comment on most lines giving the idea of what it does would be helpful to follow along for the rest of the examples.
  - I think apigatewayv1 is sometimes overboard on in-line comments. apigatewayv2 is a little underboard.

File: packages/@aws-cdk/aws-apigatewayv2-authorizers-alpha/README.md [hunk 3]
```diff
@@ -65,12 +65,12 @@ const api = new apigwv2.HttpApi(this, 'HttpApi', {
...
 ### Route Authorization
 
-Authorization can also configured for each Route. When a route authorization is configured, it takes precedence over default authorization.
+Authorization can also be configured for each Route. When a route authorization is configured, it takes precedence over default authorization.
 
 The example below showcases default authorization, along with route authorization. It also shows how to remove authorization entirely for a route.
 
 - `GET /books` and `GET /books/{id}` use the default authorizer settings on the api
-- `POST /books` will require the [write:books] scope
+- `POST /books` will require the ['write:books'] scope
```
Comment (kaizencc) [inline packages/@aws-cdk/aws-apigatewayv2-authorizers-alpha/README.md:73]:
```suggestion
- `POST /books` will require the `['write:books']` scope
```

**Review by mikewrighton — APPROVED** [review]

**Review by mikewrighton — APPROVED** [review]

**Pull request merged**